### PR TITLE
Change psdb bash script to work with symlinks

### PR DESCRIPTION
**Pull request by brkzlr**
`dirname` does not work correctly when you call the script through a symlink unless you route it first through `readlink -f`, because `dirname` will always give you the directory of the symlink itself.

An example for when this would error out: I had the psdb portable folder in `/opt/psdb` but I threw a symlink to the bash script inside `/usr/local/bin` instead of modifying `PATH` to add the opt folder.
Any psdb command would assume the bin/config folder would be in `/usr/local/bin` instead of its true location.

### CODE REVIEW

**Comment by alex-free** [comment]
Thanks, today I learned!

**Comment by alex-free** [comment]
Thanks, today I learned!

**Comment by alex-free** [comment]
@brkzlr  I just remembered why I didn't have it  do just a single `cd "$(dirname "$0")"` originally (hence same problem with one  `cd "$(dirname "$(readlink -f "$0")")"`.

When it's parsing the file, if you gave it a relative path as the input file it won't be able to find the input file, because the relative path is now from a different source. So i.e. if you did
(`$1` is for example `../somefile.bin`, so you could have `psdb ../somefile.bin` )
`cd "$(dirname "$(readlink -f "$0")")"`
`cp "$1" $tmp`
then you get an error, because the relative path's source from the original invocation is not what it is after `cd`'ing

I actually took great care to never cd into the script directory directly, but use it to derive absolute paths for this reason.

So basically, I think it's better if we replace each instance of `"$(dirname "$0")"` previously with `"$(dirname "$(readlink -f "$0")")"`

instead of doing one `cd` using it.

Thoughts?

**Comment by brkzlr** [comment]
Oh, sorry for getting back so late. Looks like you edited your double post and GitHub does not notify on post edits, even if you ping me in edit, so I didn't know you said something.
Funnily enough Discord does the same thing, it never notifies the pinged user if you ping in a post edit instead of new post.

Yeah that seems fair, didn't know about that shortcoming, although I would suggest to save the result of `"$(dirname "$(readlink -f "$0")")"` into a variable like `SCRIPTDIR` and then just reference that as the value of the variable would be unlikely to change during the script runtime.

Makes it easier to read as well.
